### DISCUSSION TITLE
LDAPS and self-signed TLS certificate

### DISCUSSION THREAD

**Discussion by js19313** [kimai/kimai#5198] [Docker] [answered]
Hi,

I'm running Kimai as docker container and have enabled user authentication via LDAP. When I switch to LDAPS, Kimai refuses the connection because my DC is using a self-signed TLS server certificate. How can I tell Kimai to trust the CA certificate that issued the LDAPS server cert? Would it be sufficient to install the CA into the ca-certificates system truststore? Or is there by chance any undocumented environment variable that can be used to pass the trusted certificate on to Kimai, or alternatively tell Kimai to ignore certificate chain validation errors?

I'm running the Kimai apache2 image v2.25.0.

Thanks!

**Comment by kevinpapst** [comment]
I personally have no clue and have never used that setup. Maybe someone else has. 

What I can share:
Kimai uses the Laminas LDAP implementation: https://docs.laminas.dev/laminas-ldap/
This here talks about the certificate: https://docs.laminas.dev/laminas-authentication/adapter/ldap/

Not sure if that helps, but I would love to add a solution to the documentation. So if you find one, please let me know!

**Reply by js19313** [reply, marked as answer]
`Alternately, if you are concerned that the server could be spoofed, you can export the LDAP server's root certificate and put it on the web server so that the OpenLDAP client can validate the server's identity.` 

Since this sounded like it would be enough to add the CA certificate to the system truststore, I ended up installing the CA certificate to the system hosting the docker image and mapping the ca-certificates.crt file into the Kimai container in the volumes section of my docker-compose file:

`- /etc/ssl/certs/ca-certificates.crt:/etc/ssl/certs/ca-certificates.crt`

Now the LDAPS connection works without issues.
Tested with both the useStartTls and useSsl parameter.

**Reply by kevinpapst** [reply]
Thanks for sharing 👍  Would you expect that in the Docker or LDAP docs?

EDIT: added it to the LDAP troubleshooting section
